internal/wasm/gofunc.go

```diff
@@ -46,6 +47,97 @@ func GetHostFunctionCallContextValue(fk FunctionKind, ctx *ModuleContext) *refle
 	return nil
 }
 
+// PopGoFuncParams pops the correct number of parameters off the stack into a parameter slice for use in CallGoFunc
```

Comment:
besides tidying up the pop loop, the main change here to logic is to avoid allocating empty slices when there are no inputs or outputs.